### ISSUE TITLE
[ER] Warn against unused braces for const value argument

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#83299]
```rust
#![allow(dead_code)]
struct Foo<const N: usize> {
    data: [u32; N],
}
impl<const N: usize> Foo<{ N }> {
}
fn main() {}
```

Gives no warnings, but I expected a warning like:

```rust
warning: unnecessary braces around const value
 --> ...\test.rs:5:26
  |
5 | impl<const N: usize> Foo<{ N }> {
  |                          ^^^^^ help: remove these braces
  |
  = note: `#[warn(unused_braces)]` on by default
```

**Comment by osa1** [comment]
Apparently this is a known issue: https://github.com/rust-lang/rust/blob/eb95acea8aeaeef834214eaffb15d64095fe9271/compiler/rustc_lint/src/unused.rs#L991 blocked on #67075.

**Comment by leonardo-m** [comment]
So do you suggest to close this issue down? Or do we keep this open to keep memory of this little problem?

**Comment by osa1** [comment]
I couldn't find an open issue for this FIXME so I think we can use this one as the tracking issue.
